docs/logger-mdc-instrumentation.md

```diff
@@ -0,0 +1,37 @@
+# Logger MDC auto-instrumentation
+
+The Mapped Diagnostic Context (MDC) is
+
+> an instrument for distinguishing interleaved log output from different sources.
+> &mdash; <cite> [log4j MDC documentation](http://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/MDC.html) </cite>
+
+It contains thread-local contextual information which is later copied to each logging event captured
+by a logging library.
+
+The OTel Java agent injects several pieces of information about the current span into each logging
+event's MDC copy:
+
+- `traceId` - the current trace id
+  (same as `TracingContextUtils.getCurrentSpan().getContext().getTraceIdAsHexString()`);
+- `spanId` - the current span id
+  (same as `TracingContextUtils.getCurrentSpan().getContext().getSpanIdAsHexString()`);
+- `sampled` - a boolean flag marking whether the current span is sampled or not
+  (same as `TracingContextUtils.getCurrentSpan().getContext().isSampled()`).
+  Only added when `sampled` is `true`.
```

Comment:
If it is always true, why include it?

Reply:
I've added an explanation to the doc. Basically if it's there the span is sampled, if not it is not - I guess the reasoning behind it is that most of spans are not sampled and keeping sampled=false in every log line would be just a waste of space.

Reply:
Aren't most span sampled? With `AlwaysOn` sampler, which is default

Reply:
🤦  That's true...

I somehow thought that it worked completely the other way 😅 

Now I'm starting to wonder why it works that way... @anuraaga I think you've added it first in the logback instrumentation -- why do we only log `true` value?